### PR TITLE
test: improve unit test coverage of checker

### PR DESCRIPTION
**Pull request by GMHDBJD**
### What problem does this PR solve? 
part of #998 

### Check List 

Tests 

 - Unit test

### CODE REVIEW

File: checker/check_test.go [hunk 1]
```diff
@@ -26,11 +31,48 @@ func TestChecker(t *testing.T) {
...
+		config.ReplicationPrivilegeChecking,
+		config.VersionChecking,
+		config.BinlogEnableChecking,
+		config.BinlogFormatChecking,
+		config.BinlogRowImageChecking,
+		config.TableSchemaChecking,
+		config.ShardTableSchemaChecking,
+		config.ShardAutoIncrementIDChecking,
+	}
+	ignoreCheckingItems := make([]string, 0)
```
Comment (lance6716) [inline checker/check_test.go:66]: `make` for some capacity?

Comment (lance6716) [inline checker/check_test.go:66]: `make` for some capacity or no zero len?

**Review by lance6716 — COMMENTED** [review]
rest LGTM

File: checker/check_test.go (hunk 1 above)
Comment (GMHDBJD) [reply checker/check_test.go]: address in 8ac2fd6

Comment (lance6716) [reply checker/check_test.go]: could use zero length and this capacity.

File: checker/check_test.go [hunk 2]
```diff
@@ -40,3 +82,294 @@ func (t *testChecker) TestCheckSyncConfig(c *tc.C) {
...
+					UNIQUE KEY u_b(b)
+					) ENGINE=InnoDB DEFAULT CHARSET=latin1`
+
+	mock := s.initMockDB(c)
+	mock.ExpectQuery("SHOW DATABASES").WillReturnRows(sqlmock.NewRows([]string{"DATABASE"}).AddRow(schema))
+	mock.ExpectQuery("SHOW FULL TABLES").WillReturnRows(sqlmock.NewRows([]string{"Tables_in_" + schema, "Table_type"}).AddRow(tb1, "BASE TABLE").AddRow(tb2, "BASE TABLE"))
+	mock.ExpectQuery("SHOW VARIABLES LIKE 'sql_mode'").WillReturnRows(sqlmock.NewRows([]string{"Variable_name", "Value"}).AddRow("sql_mode", ""))
+	mock.ExpectQuery("SHOW CREATE TABLE .*").WillReturnRows(sqlmock.NewRows([]string{"Table", "Create Table"}).AddRow(tb1, fmt.Sprintf(createTable1, tb1)))
+	mock.ExpectQuery("SHOW CREATE TABLE .*").WillReturnRows(sqlmock.NewRows([]string{"Table", "Create Table"}).AddRow(tb1, fmt.Sprintf(createTable1, tb2)))
+	c.Assert(CheckSyncConfig(context.Background(), cfgs), tc.ErrorMatches, "(.|\n)*same table name in case-sensitive(.|\n)*")
```
Comment (csuzhangxc) [inline checker/check_test.go:374]: Should the error message should be `case-insensitive`?

**Review by csuzhangxc — COMMENTED** [review]
rest LGTM

File: checker/check_test.go (hunk 1 above)
Comment (GMHDBJD) [reply checker/check_test.go]: Stupid me！😂

**Review by csuzhangxc — COMMENTED** [review]
LGTM

**Comment by lance6716** [comment]
LGTM

**Review by lance6716 — APPROVED** [review]

**Pull request merged**